sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -911,19 +911,30 @@ class Analyzer(
      * Resolve the correlated expressions in a subquery by using the an outer plans' references. All
      * resolved outer references are wrapped in an [[OuterReference]]
      */
-    private def resolveOuterReferences(plan: LogicalPlan, outer: LogicalPlan): LogicalPlan = {
+    private def resolveOuterReferences(
+        plan: LogicalPlan,
+        outers: Seq[LogicalPlan]): LogicalPlan = {
+      var throwError = false
       plan transformDown {
         case q: LogicalPlan if q.childrenResolved && !q.resolved =>
           q transformExpressions {
             case u @ UnresolvedAttribute(nameParts) =>
               withPosition(u) {
                 try {
-                  outer.resolve(nameParts, resolver) match {
+                  val outer = outers.iterator
```

Comment:
Is there are way to simplify the code below? Two try's seem a bit hacky? We could move try catch into the loop and throw an exception if we could not resolve the expr....